Dockerfile

```diff
@@ -2,9 +2,21 @@ FROM python:2.7
 
 ADD . /src
 WORKDIR /src
-RUN pip install -r requirements.txt
+RUN pip install -r requirements.txt -e .
 RUN python setup.py develop
 
+# Install Custodian Azure
+WORKDIR /src/tools/c7n_azure
+RUN pip install -r requirements.txt -e .
+RUN python setup.py develop
+
+# Install Custodian GCP
+WORKDIR /src/tools/c7n_gcp
+RUN pip install -r requirements.txt -e .
+RUN python setup.py develop
+
+RUN pip install future
```

Comment:
fwiw, the c7n-org tools has a good canonical dockerfile, the future install here looks odd.

Reply:
Thanks! I looked into the c7n-org dockerfile. Looks like c7n_azure needs this module which isn't present in any of the base docker images i.e.(pypy2 or python2.7). This is the error that pops up:
```Traceback (most recent call last):
  File "/usr/local/bin/custodian", line 11, in <module>
    load_entry_point('c7n', 'console_scripts', 'custodian')()
  File "/src/c7n/cli.py", line 368, in main
    command(config)
  File "/src/c7n/commands.py", line 349, in schema_cmd
    load_resources()
  File "/src/c7n/resources/__init__.py", line 99, in load_resources
    resources.load_plugins()
  File "/src/c7n/registry.py", line 117, in load_plugins
    f = ep.load()
  File "/usr/local/site-packages/pkg_resources/__init__.py", line 2346, in load
    return self.resolve()
  File "/usr/local/site-packages/pkg_resources/__init__.py", line 2352, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/src/tools/c7n_azure/c7n_azure/entry.py", line 15, in <module>
    from c7n_azure.provider import resources
  File "/src/tools/c7n_azure/c7n_azure/provider.py", line 20, in <module>
    from .session import Session
  File "/src/tools/c7n_azure/c7n_azure/session.py", line 27, in <module>
    from c7n_azure.utils import ResourceIdParser, StringUtils
  File "/src/tools/c7n_azure/c7n_azure/utils.py", line 20, in <module>
    from builtins import bytes
ImportError: No module named builtins```

Installing the future module fixes this. Can we put it out in the requirements.txt file?

Reply:
thats odd thats a python3 only thing surprised that doesn't fail with py2 on ci

Reply:
c7n_azure packages already has a dep on futures in its setup.py